### PR TITLE
Support multi-provider MCs

### PR DESCRIPTION
**Pull request by QuentinBisson**
Towards https://github.com/giantswarm/giantswarm/issues/29703

This needs to be tested but this PR adds support for multi-provider MC. It basically configure the provider label depending on the actual infrastructure deployed. We keep it the same for vintage MCs

## Checklist

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: service/key/key.go
Comment (hervenicol) [inline service/key/key.go:143]: Is the `²` really part of the `Kind` name?

File: service/key/key.go
Comment (hervenicol) [inline service/key/key.go:147]: What are the cases where we could fall back to this default behaviour? I mean, is it expected that some clusters would not have a `provider.Kind`?
If this is not expected, maybe we should rather set `Unknown` so we can detect this more easily?

File: service/key/key.go
Comment (QuentinBisson) [reply service/key/key.go:143]: Nope, that's a typo

Comment (hervenicol) [reply service/key/key.go:143]: Then:
```suggestion
		case "GCPManagedCluster":
```

File: service/key/key.go
Comment (QuentinBisson) [reply service/key/key.go]: So it could technically happen for vintage or if that value was not set by the customer so erroring out makes sense :)

**Comment by QuentinBisson** [comment]
@hervenicol thanks for the review comments, I'll test it on monday on a vintage MC, CAPI MC and multi-provider one :)

**Comment by QuentinBisson** [comment]
Tested on Gauss, Goat and Glean;

VSphere cluster:
image

CAPZ cluster:
image

**Comment by QuentinBisson** [comment]
Waiting for review from @giantswarm/team-atlas 

We might want to do something about the region label but it does not strike me as an issue for now

File: service/controller/resource/monitoring/prometheus/test/case-5-eks-v18.golden
Comment (hervenicol) [reply service/controller/resource/monitoring/prometheus/test/case-5-eks-v18.golden:41]: The tests look odd.
For instance, we have some file named `./service/controller/resource/monitoring/scrapeconfigs/test/openstack/case-2-aws-v16.golden`.
It used to set `replacement: openstack`, but now sets `replacement: capa`.
I don't really understand how these tests work, but are they supposed to simulate a `openstack` config, a vintage `aws` config, on a `capa` config?

Comment (QuentinBisson) [reply service/controller/resource/monitoring/prometheus/test/case-5-eks-v18.golden:41]: There is a big issue with those tests now :( 

First, openstack should be removed but we now set replacement: capa because the cluster CR reference an AWSCluster CR. 

For the eks case, the issue is that our test actually renders an EKS (AWSManagedCluster) cluster on a Vintage MC so the condition is wrong from the start ...

Comment (QuentinBisson) [reply service/controller/resource/monitoring/prometheus/test/case-5-eks-v18.golden:40]: I think the main issue is that the tests are generally Vintage focused more than CAPI focused :(

Comment (QuentinBisson) [reply service/controller/resource/monitoring/prometheus/test/case-5-eks-v18.golden:40]: For this use case, we go through this case https://github.com/giantswarm/prometheus-meta-operator/blob/92f9a43b250ca401efbe03e5978bbeee453c0006/service/key/key.go#L124 (IsVintageManagementCluster because of this https://github.com/giantswarm/prometheus-meta-operator/blob/92f9a43b250ca401efbe03e5978bbeee453c0006/service/controller/resource/monitoring/prometheus/resource_test.go#L81)

Comment (hervenicol) [reply service/controller/resource/monitoring/prometheus/test/case-5-eks-v18.golden:40]: I was mostly concerned that the "broken" tests show broken logic in the code.

If we're sure that the problem comes from the test, we could probably merge now without fixing the tests. But then, I'm not sure we can spot future mistakes with the tests.
So, I think we should fix the tests. But if it's in a separate PR that looks fine to me.

**Review by hervenicol — APPROVED** [review]
Apart from the tests, code looks ok to me.

**Comment by QuentinBisson** [comment]
Merging this and we will fix the tests before releasing :)

**Pull request merged**
